open-craft/xblock-html
# [BB-6212] Add themeable Tables to the TinyMCE Editor and use platform bundle for tinymce

## Description

This PR brings the following changes to the XBlock

* This adds the table plugin the studio TinyMCE editor and updates the
  bleaching module to allow tables to be rendered in the output
* Adds a Custom Theme for the table plugin which will insert the class
  "themed-table" to the tables created in the TinyMCE editor. This can
  be used by the clients to apply styles to the table based on their
  theme guidelines
* Sets the prefix to the static files using django settings.STATIC_URL
  to allow proper prefixing of the static assets imported from platform
* Updates the bleach library to version 5.0.0 and makes corresponding
  changes in bleaching.py

## Testing Instructions

1. Checkout `edx-platform` to the `open-craft:DubeySandeep/tinymce_upgrade` branch (ref: https://github.com/openedx/edx-platform/pull/30335/)
2. Install the XBlock from this PR Branch
3. Configure the XBlock in "Advanced Settings" of the test course - add "html5" and "excluded_html5" to the advanced modules
4. Edit a Unit
5. Click Advanced -> Text
6. Click the Edit button of the added block, this should bring up the TinyMCE Editor. Along with other editor icons, there should be Table related icons as well.
7. Click the "Add Table" ICON to insert a new table (you will have to select the size of the table). Enter some values.
8. Right Click on the table and go to table properties -> Select Class => ".themed-table". You can set other styles like background, border colors to verify those don't have any effect on the output.
9. Save the Block.
10. Right click on the table content and navigate to the Table DOM element.
11. Verify that the "themed-table" class is present in the table and the "styles" are stripped off. This allows only the organization's themes styling to apply for the table.

### Screenshots

#### Setting the custom class

![image](https://user-images.githubusercontent.com/383862/176698653-fe19bc93-d3dc-471e-88c3-c030e9cfd812.png)

**Note:** The unnecessary title `Custom Theme` shown in the screenshot was removed later for simplicity.

#### Rendered Table with no styles and only the custom class

![image](https://user-images.githubusercontent.com/383862/176699356-4266e688-624d-4b26-afd6-fb63b38851f7.png)


### Related information

* [OpenCraft JIRA Ticket](https://tasks.opencraft.com/browse/BB-6212)
